### ISSUE TITLE
Implement Machine Learning Pipeline

### ISSUE THREAD

**Issue by hershal** [hershic/cs231n#8]
Dependencies:
- [ ] backprop
- [ ] gradient descent
- [ ] softmax
